docs/production/deployment.md

```diff
@@ -337,6 +337,12 @@ server {
                 proxy_buffering         off;
                 proxy_read_timeout      20m;
                 proxy_pass              https://zulip-upstream-host;
+
+                # Should you use an http instead of an https URL for the upstream host,
+                # then uncomment the line below and add "USE_X_FORWARDED_PORT = True"
+                # to your settings.py as well:
```

Comment:
Is this an issue any time the ports are different, with HTTP vs HTTPS just being one specific reason this might happen?

Reply:
As far as I understand it, it also happens when the internal port differs from the external one. Then the reverse function in Django generates an incorrect URL: https://docs.djangoproject.com/en/3.2/ref/urlresolvers/

Reply:
Yeah, I think the more accurate comment would be:
```
                # If the zulip upstream host is not listening on port 443 (e.g. it is using
                # http, or has set `nginx_listen_port`), then uncomment the line below and
                # add "USE_X_FORWARDED_PORT = True" to your settings.py as well:
```

We should also add equivalent mentions for the Apache2 and HAProxy sections.  For Apache2, that's:
```
RequestHeader set "X-Forwarded-Port" "443"
```

Tangentially, I don't think `X-Forwarded-SSL` (in the Apache2 headers) does anything, and we should probably remove it.